Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -199,14 +219,39 @@ lemma IsIntegralCurveOn.comp_add (hγ : IsIntegralCurveOn γ v s) (dt : ℝ) :
   simp only [mfld_simps, hasFDerivWithinAt_univ]
   exact HasFDerivAt.add_const (hasFDerivAt_id _) _
 
-lemma isIntegralCurveOn_comp_add {dt : ℝ} :
+lemma isIntegralCurveOn_comp_add (dt : ℝ) :
     IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· + dt)) v { t | t + dt ∈ s } := by
   refine ⟨fun hγ ↦ hγ.comp_add _, fun hγ ↦ ?_⟩
   convert hγ.comp_add (-dt)
   · ext t
     simp only [Function.comp_apply, neg_add_cancel_right]
   · simp only [mem_setOf_eq, neg_add_cancel_right, setOf_mem_eq]
 
+lemma isIntegralCurveOn_comp_sub (dt : ℝ) :
+    IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· - dt)) v { t | t - dt ∈ s } := by
+  refine ⟨fun hγ ↦ hγ.comp_add _, fun hγ ↦ ?_⟩
+  convert hγ.comp_add dt
+  ext t
+  · simp only [Function.comp_apply, add_sub_cancel]
+  · simp only [mem_setOf_eq, add_sub_cancel, setOf_mem_eq]
```

Comment:
What you have is fine but the convert-ext-simp pattern is so common that I like the make it easy to spot at a glance:
```suggestion
  convert hγ.comp_add dt <;> ext <;> simp
```